eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -121,10 +119,117 @@ function Confirm-ChangeLogEntry {
   return $true
 }
 
-function Set-TestChangeLog($TestVersion, $changeLogFile, $ReleaseEntry) {
-  Set-Content -Path $changeLogFile -Value @"
-# Release History
-## $TestVersion ($(Get-Date -f "yyyy-MM-dd"))
-- $ReleaseEntry
-"@
+function New-ChangeLogEntry {
+  param (
+    [Parameter(Mandatory = $true)]
+    [ValidateNotNullOrEmpty()]
+    [String]$Version,
+    [String]$Status="(Unreleased)",
+    [String]$Title,
+    [String[]]$Content
+  )
+
+  # Validate relase Status
+  $Status = $Status.Trim().Trim("()")
+  if ($Status -ne "Unreleased") {
+    $dateFormat = "yyyy-MM-dd"
+    $provider = [System.Globalization.CultureInfo]::InvariantCulture
+    try {
+      $Status = ([System.DateTime]::ParseExact($Status, $dateFormat, $provider)).ToString($dateFormat)
+    }
+    catch {
+        LogWarning "Invalid date [ $Status ] passed as status for Version [$Version]. Please use a valid date in the format '$dateFormat' or use '(Unreleased)'"
+    }
+  }
+
+  $Status = "($Status)"
```

Comment:
Why add the parens `()` here? It seems like you have to remove them in various places. Maybe you should only add them when you are building the title.

Reply:
So far we have been adding the parenthesis.  I only removed it to verify the date. Ad that also take care of cases where it was not added.

Reply:
I think its better to continue with this, or else I will have to update a lot of scripts where we are already consuming the $ReleaseStatus with the parenthesis.

Reply:
Are there scripts outside of the scope of this PR that would need to be changed? If not then I still think that it might be better to make the change and assume there is no parenthesis around the status as you are removing and adding in various places which makes it complex to understand when they should be or should not be there.

Reply:
Yes there are a couple of scripts that are already using this. Specifically the new collect changelog scripts in the language repos.

Reply:
OK that make this a leaky abstraction because now all consumers have to know about and handle these parenthesis. Perhaps we can consider improvements when we work on those scripts.